VirtoCommerce.Storefront/Controllers/Api/ApiAccountController.cs

```diff
@@ -430,5 +431,59 @@ public async Task<ActionResult> UpdateAddresses([FromBody] IList<Address> addres
 
             return Ok();
         }
+
+        // DELETE: storefrontapi/account/phonenumber
+        [HttpDelete("phonenumber")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult<RemovePhoneNumberResult>> RemovePhoneNumber()
+        {
+            var result = await _signInManager.UserManager.SetPhoneNumberAsync(WorkContext.CurrentUser, null);
+            await _signInManager.SignInAsync(WorkContext.CurrentUser, isPersistent: false);
+
+            return new RemovePhoneNumberResult { Succeeded = result.Succeeded, Errors = result.Errors.Select(x => x.Description) };
+        }
+
+        // POST: storefrontapi/account/twofactorauthentification
+        [HttpPost("twofactorauthentification")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult<ChangeTwoFactorAuthenticationResult>> EnableTwoFactorAuthentication([FromBody] EnableTwoFactorAuthenticationModel model)
+        {        
+            var phoneConfirmed = await _signInManager.UserManager.IsPhoneNumberConfirmedAsync(WorkContext.CurrentUser);
+            if (!phoneConfirmed)
+            {
+                var url = "/account/phonenumber";
+
+                return new ChangeTwoFactorAuthenticationResult { Succeeded = false, VerificationUrl = url };
+            }
+                             
+            var result = await _signInManager.UserManager.SetTwoFactorEnabledAsync(WorkContext.CurrentUser, !model.Enabled);
```

Comment:
Why **!** model.Enabled (NOT is the problem) - could we use value we get from model? Probably we could change it when passing from JS.